person.rb

```diff
@@ -1,20 +1,30 @@
 require_relative 'nameable'
+require_relative 'book'
+require_relative 'rental'
+
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(id, age = 0, name: 'Unknown', parent_permission: true)
     super(id)
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def rent_book(book, date)
+    rental = Rental.new(date, self, book)
+    @rentals << rental
+    rental
+  end
```

Comment:
- [OPTIONAL] You have done well implementing this method as requested by the previous reviewer, I'll suggest changing the name from `rent_book` to ``add_rental`` as this method adds new rental and this seems like a more appropriate name for it. :+1: :smile: